api/src/main/java/io/strimzi/api/kafka/model/KafkaMirrorMaker2ConnectorSpec.java

```diff
@@ -14,8 +15,20 @@
         builderPackage = Constants.FABRIC8_KUBERNETES_API
 )
 @JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonPropertyOrder({"tasksMax", "config"})
+@JsonPropertyOrder({"tasksMax", "autoRestartConnectorsAndTasks", "config"})
 @EqualsAndHashCode(callSuper = true)
 public class KafkaMirrorMaker2ConnectorSpec extends AbstractConnectorSpec {
     private static final long serialVersionUID = 1L;
+
+    private AutoRestart autoRestartConnectorsAndTasks = new AutoRestart();
```

Comment:
I think we deviate here from the proposal. The proposal suggests to have it once per `KafkaMirrorMaker2` CR - so that would be added to `KafkaMirrorMaker2Spec` class. But here you put it per connector.

In general, I do not have problem with that. But:
* It should be discussed and made clear since it deviates from the approved proposal
* If this is the way we want to go, then you can make the code simple and just have the
  ```yaml
   autoRestart:
     enabled: false
  ```
  directly in the `AbstractConnectorSpec` and use the same APi for both of them.

CC @strimzi/maintainers @ajborley

Reply:
I don't have a problem with that either - I think it's unlikely that a user would want to switch autoRestart off for one MM2 connector and not the others, but at least this way they have the option.

(thanks for tagging me in here Jakub, and thanks @ThomasDangleterre for picking this up! :) )